server/webapp/WEB-INF/rails.new/spec/new_javascripts/pipeline_configs/views/materials_config_widget_spec.js

```diff
@@ -66,7 +108,20 @@ define(["jquery", "mithril", "pipeline_configs/models/materials", "pipeline_conf
         filter:      new Materials.Filter({ignore: ['*.doc']})
       });
 
-      tfsMaterial = materials.createMaterial({
+      mount(materials);
+      viewMaterial();
+
+      verify(perforceMaterial);
+      expect($root.find("input[data-prop-name='port']").val()).toBe(perforceMaterial.port());
+      expect($root.find("input[data-prop-name='username']").val()).toBe(perforceMaterial.username());
+      expect($root.find("input[data-prop-name='view']").val()).toBe(perforceMaterial.view());
+      expect($root.find("input[data-prop-name='passwordValue']").val()).toBe('p@ssw0rd');
+      expect($root.find("input[data-prop-name='useTickets']").val()).toBe('on');
+    });
```

Comment:
I think we should avoid clubbing all the expectations together. Failures become notoriously hard to locate without good descriptions.
You can nest the describes

```
describe("mercurial component", function () {
 beforeEach() {
  mount(materials);
 }
 describe("model binding" , function(){
    it("should bind url" , function(){
       expect($root.find("input[data-prop-name='url']").val()).toBe(mercurialMaterial.url());
     });
   });
}
```

http://jasmine.github.io/2.0/introduction.html#section-Nesting_<code>describe</code>_Blocks